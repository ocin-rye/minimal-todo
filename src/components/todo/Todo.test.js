import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Todo from './index';

describe('<Todo />', () => {
  const mockToggleCompleted = jest.fn();
  const mockDelete = jest.fn();

  describe('Active todo', () => {
    const todoActive = {
      completed: false,
      id: 'SGcqcq0TbTPqvyWOMIKtZ',
      text: 'Example Todo Text.',
    };

    const todo = shallow(
      <Todo todo={todoActive} toggleTodoCompletion={mockToggleCompleted} deleteTodo={mockDelete} />,
    );

    it('renders active todo without crashing', () => {
      expect(toJson(todo)).toMatchSnapshot();
    });

    it('calls toggle completed fn() on click', () => {
      todo.find('.toggleCompleted').simulate('click');
      expect(mockToggleCompleted).toHaveBeenCalled();
    });

    it('calls delete fn() on click', () => {
      todo.find('.delete').simulate('click');
      expect(mockDelete).toHaveBeenCalled();
    });
  });

  describe('Completed todo', () => {
    const todoCompleted = {
      completed: true,
      id: 'SGcqcq0TbTPqvyWOMIKtZ',
      text: 'Example Todo Text.',
    };

    const todo = shallow(
      <Todo
        todo={todoCompleted}
        toggleTodoCompletion={mockToggleCompleted}
        deleteTodo={mockDelete}
      />,
    );

    it('renders completed todo without crashing', () => {
      expect(toJson(todo)).toMatchSnapshot();
    });

    it('calls toggle completed fn() on click', () => {
      todo.find('.toggleCompleted').simulate('click');
      expect(mockToggleCompleted).toHaveBeenCalled();
    });

    it('calls delete fn() on click', () => {
      todo.find('.delete').simulate('click');
      expect(mockDelete).toHaveBeenCalled();
    });
  });
});
