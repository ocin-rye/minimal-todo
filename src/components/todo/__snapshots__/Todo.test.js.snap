// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> Active todo renders active todo without crashing 1`] = `
<div
  className="todo"
>
  <p
    className=""
  >
    Active
  </p>
  <p
    className=""
  >
    Example Todo Text.
  </p>
  <div
    className="actions"
  >
    <p
      className="action toggleCompleted"
      onClick={[Function]}
    >
      Mark as 
      Completed
    </p>
    <p
      className="textRight action delete"
      onClick={[Function]}
    >
      Delete
    </p>
  </div>
</div>
`;

exports[`<Todo /> Completed todo renders completed todo without crashing 1`] = `
<div
  className="todo"
>
  <p
    className=""
  >
    Completed
  </p>
  <p
    className=""
  >
    Example Todo Text.
  </p>
  <div
    className="actions"
  >
    <p
      className="action toggleCompleted"
      onClick={[Function]}
    >
      Mark as 
      Active
    </p>
    <p
      className="textRight action delete"
      onClick={[Function]}
    >
      Delete
    </p>
  </div>
</div>
`;
